demo/project_demo12-HopsworksTikTokRecSys/1_pipeline.py

```diff
@@ -0,0 +1,143 @@
+from feldera import FelderaClient, SQLContext, SQLSchema
+from feldera.formats import JSONFormat, JSONUpdateFormat
+import config
+
+client = FelderaClient("http://localhost:8080")
+sql = SQLContext("mil", client)
+
+sql.register_table("interactions",
+    SQLSchema({
+        "interaction_id": "STRING",
+        "user_id": "STRING",
+        "video_id": "STRING",
+        "category_id": "INT",
+        "interaction_type": "STRING",
+        "watch_time": "INT",
+        "interaction_date": "TIMESTAMP LATENESS INTERVAL '10' SECONDS",
+        "previous_interaction_date": "TIMESTAMP",
+        "interaction_month": "TIMESTAMP",
+    })
+)
+
+sql.register_local_view("interaction_type_groupby_video_id_count", """
+    SELECT
+        video_id,
+        interaction_type,
+        count(interaction_type) as interaction_type_count
+    FROM
+        interactions
+    GROUP BY (interaction_type, video_id)
+""")
+
+sql.register_local_view("interaction_type_groupby_user_id_count", """
+    SELECT
+        user_id,
+        interaction_type,
+        count(interaction_type) as interaction_type_count
+    FROM
+        interactions
+    GROUP BY (interaction_type, user_id)
+""")
+
+sql.register_local_view("video_id_with_interactions_type_pivot", """
+    SELECT
+        *
+    FROM
+        interaction_type_groupby_video_id_count
+    PIVOT (
+        SUM(interaction_type_count) FOR interaction_type IN (
+            'like' as like_ct,
+            'dislike' as dislike_ct,
+            'view' as view_ct,
+            'comment' as comment_ct,
+            'share' as share_ct,
+            'skip' as skip_ct
+        )
+    )
+""")
+
+sql.register_local_view("video_id_with_interactions_type_pivot", """
```

Comment:
I think this view is accidentally created twice.

Reply:
Oh yes, its a duplicate.